src/hostManager/clientManager.ts

```diff
@@ -0,0 +1,189 @@
+import { Context } from '@/enums/context.js';
+
+/**
+ * subpath per context
+ * it should not have a trailing slash
+ */
+const CONTEXT_PATHS = new Map<Context, string>([
+  [Context.Builder, '/builder'],
+  [Context.Player, '/player'],
+  [Context.Auth, '/auth'],
+  [Context.Account, '/account'],
+  [Context.Analytics, '/analytics'],
+]);
+
+export class ClientManager {
+  private static INSTANCE: ClientManager | null;
+  private host: URL;
+  private readonly clientHosts = new Map<Context, URL>();
+
+  /**
+   * defined prefixes for items, that should have trailing slash
+   * we need to define library's prefix from the client
+   */
+  private readonly itemPrefixes = new Map<Context, string>([
+    [Context.Builder, 'items/'],
+    [Context.Player, ''],
+    [Context.Library, 'collections/'],
+    [Context.Analytics, 'items/'],
+  ]);
+
+  private constructor() {
+    try {
+      this.host = new URL(window.location.href);
+    } catch {
+      this.host = new URL('http://mock.graasp.org/');
+    }
+  }
+
+  public static getInstance() {
+    if (!this.INSTANCE) {
+      this.INSTANCE = new ClientManager();
+      ClientManager.getInstance();
+    }
+    return this.INSTANCE;
+  }
+
+  /**
+   * Define default host
+   * @param host URL that should have a trailing slash
+   * @returns instance
+   */
+  public setHost(host: URL) {
+    this.host = host;
+    return this;
+  }
+
+  /**
+   * Add a specific host for given context. A context cannot have its host set twice.
+   * @param {Context} context
+   * @param {URL} host
+   * @returns instance
+   */
+  public addHost(context: Context, host: URL) {
+    if (this.clientHosts.has(context)) {
+      throw new Error(
+        `The given context '${context}' is already present in the hosts.`,
+      );
+    }
+
+    this.clientHosts.set(context, host);
```

Comment:
Since host is a URL object, should we make an explicit copy so we do not risk inadvertent mutations from outside components ?
Below is an example of the issue we could face:
<img width="297" alt="Screenshot 2025-02-03 at 11 00 34" src="https://github.com/user-attachments/assets/21386bbc-69b5-4ade-bec0-84da0c8f479d" />